src/algebra/category/Group/images.lean

```diff
@@ -0,0 +1,99 @@
+/-
+Copyright (c) 2020 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+-/
+import algebra.category.Group
+import category_theory.limits.shapes.images
+import group_theory.subgroup
+import algebra.group_power
+
+/-!
+# The category of commutative additive groups has images.
+-/
+
+open category_theory
+open category_theory.limits
+
+universe u
+
+namespace AddCommGroup
+
+variables {G H : AddCommGroup.{0}} (f : G ⟶ H)
+
+local attribute [ext] subtype.eq'
+
+section -- implementation details of `has_image` for AddCommGroup; use the API, not these
+/-- the image of a morphism in AddCommGroup is just the bundling of `set.range f` -/
+def image : AddCommGroup := AddCommGroup.of (set.range f)
+/-- the inclusion of `image f` into the target -/
+-- TODO it would be nicer to reuse existing unbundled machinery here: does it exist?
+def image.ι : image f ⟶ H := (⟨subtype.val, rfl, by { intros, refl }⟩ : image f →+ H)
+instance : mono (image.ι f) :=
+begin
+  split, intros, ext,
+  convert congr_arg (λ k : Z ⟶ H, (k : _ →+ _) x) w,
+end
+/-- the corestriction map to the image -/
+-- TODO it would be nicer to reuse existing unbundled machinery here: does it exist?
```

Comment:
Can't you reuse `range_factorization`, like you did for `Type`?

Reply:
Ah, yes. I've made another commit reusing more machinery.